src/bundle/Resources/public/js/scripts/admin.notifications.modal.js

```diff
@@ -1,89 +1,171 @@
-            event.target.closest(SELECTOR_MODAL_ITEM).querySelector(SELECTOR_DESC_TEXT).classList.remove(CLASS_ELLIPSIS);
+            event.target
+                .closest(SELECTOR_MODAL_ITEM)
+                .querySelector(SELECTOR_DESC_TEXT)
+                .classList.remove(CLASS_ELLIPSIS);
 
             return;
         }
 
-        markAsRead(event.target.closest(SELECTOR_MODAL_ITEM));
+        const notification = event.target.closest(SELECTOR_MODAL_ITEM);
+
+        if (!notification) {
+            return;
+        }
+
+        handleNotificationClick(notification);
     };
-    const getNotificationsStatus = () => {
-        const requestOptions = {
+    const updateNotificationsStatus = () => {
+        const notificationsTable = modal.querySelector(SELECTOR_TABLE);
+        const notificationsStatusLink = notificationsTable.dataset.notificationsCount;
+        const request = new Request(notificationsStatusLink, {
             headers: {
-                'X-CSRF-Token': token
+                'X-CSRF-Token': token,
             },
             mode: 'cors',
-            credentials: 'same-origin'
-        };
+            credentials: 'same-origin',
+        });
 
-        fetch(modal.querySelector(SELECTOR_TABLE).dataset.notificationsCount, requestOptions)
-            .then(response => response.json())
-            .then(data => setPendingNotificationCount(data));
+        fetch(request)
+            .then(handleRequestResponseJson)
+            .then(setPendingNotificationCount)
+            .catch(handleResponseError);
+    };
+    const updateUserAvatar = (notificationsInfo) => {
+        const pendingNotificationsExist = notificationsInfo.pending;
+        const userAvatar = doc.querySelector('.ez-user-menu__avatar-wrapper');
+
+        userAvatar.dataset.count = notificationsInfo.pending;
+        userAvatar.classList.toggle('n-pending-notifications', pendingNotificationsExist);
+
+        doc.querySelector('.ez-user-menu__item--notifications').dataset.count = `(${notificationsInfo.pending})`;
+    };
+    const setPendingNotificationCount = (notificationsInfo) => {
+        updateUserAvatar(notificationsInfo);
+
+        const notificationsTable = modal.querySelector(SELECTOR_TABLE);
+        const notificationsTotal = notificationsInfo.total;
+        const notificationsTotalOld = parseInt(notificationsTable.dataset.notificationsTotal, 10);
+
+        if (notificationsTotal !== notificationsTotalOld) {
+            notificationsTable.dataset.notificationsTotal = notificationsTotal;
+
+            refreshNotificationsPage();
+        }
     };
-    const getNotificationsList = () => {
-        const requestOptions = {
+    const setNotificationPage = (pageHtml) => {
+        const modalResults = modal.querySelector(SELECTOR_MODAL_RESULTS);
+
+        modalResults.innerHTML = pageHtml;
+        toggleLoading(true);
+    };
+    const toggleSpinner = (force) => {
```

Comment:
What if `force` is undefined?

Reply:
This undefined value would be passed to `toggle` function and it should be OK.
Do you want me to change it? Better solution does not come to my mind.

Reply:
Isn't `undefined` treated as `false`?

Reply:
I checked it and undefined is treated as if there was no 2 argument in `toggle`. 
Nevertheless, I noticed that in `toggleLoading` there is:
```js
toggleResults(!force);
```
which results in `true` if `force` is undefined, I will modify `toggleLoading`.

Reply:
@dew326 I have chosen `force` because in MDN in `toggle` definition there is `force`: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Reply:
Yes but when you calling method `toggleSpinner` it is more natural to do `show/hide` not to `force`